Add file extension when using `absoluteRuntime`

mbehzad opened the pull request:
es module imports need the file extension (e.g. `import "@babel/runtime/helpers/jsx.js"`, Or the filenames being listed in the package.json's subpath exports (e.g. `import "@babel/runtime/helpers/jsx"` + pkg: `"./helpers/jsx": "./helpers/jsx.js"`). when the user passes a path via `absoluteRuntime` then the rendered require statement is not the module name + subpath which will be resolved via pkg.json but rather the absolute path to the file. for this case, add the file extension / index.js to prevent bundlers from raising a warning.

<!--
Before making a PR, please read our contributing guidelines
https://github.com/babel/babel/blob/main/CONTRIBUTING.md

Please note that the Babel Team requires two approvals before merging most PRs.

For issue references: Add a comma-separated list of a [closing word](https://help.github.com/articles/closing-issues-via-commit-messages/) followed by the ticket number fixed by the PR. (it should be underlined in the preview if done correctly)

If you are making a change that should have a docs update: submit another PR to https://github.com/babel/website
-->

| Q                        | A <!--(Can use an emoji 👍) -->
| ------------------------ | ---
| Fixed Issues?            | Fixes #12824  <!-- remove the (`) quotes and write "Fixes" before the number to link the issues -->
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes (modified existing ones)
| Documentation PR Link    |  No<!-- If only readme change, add `[skip ci]` to your commits -->
| Any Dependency Changes?  | No
| License                  | MIT

<!-- Describe your changes below in as much detail as possible -->


<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12827"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

nicolo-ribaudo commented:
@mbehzad Sorry for the long delay before reviewing this PR: there have been a lot of `@babel/runtime` changes between 7.13.0 and 7.13.7 so we have been waiting to review this one.

Could you rebase this PR, and add a test with `absoluteRuntime: true` + `useESModules: true`? :pray:

mbehzad commented:
Hi @nicolo-ribaudo , I've added my changes to the current code, and added a test for `absoluteRuntime` + `useESModules` option.
**But** the regenerator import statement injection seems to have been moved to the `babel-plugin-polyfill-regenerator` module. It needs to have the `/index.js` at the end for the `absoluteRuntime` case. I'm wondering if i should add similar to the `ext` property in the runtime compat option, a flag/property to add that. or even use the `ext` prop for this and call `/index.js` the "extension" here. Let me know what is preferred. Thanx.

charlessuh commented:
@mbehzad Hi, I'm just another random person who also ran into this issue, but could you rebase this?

I'm wondering if perhaps the regenerator issue can be resolved later as a separate PR, as it may require a change to a package that lives in another repo (https://github.com/babel/babel-polyfills)

nicolo-ribaudo commented:
I have rebased https://github.com/babel/babel-polyfills/pull/79. As soon as it's merged, I'll rebase this one.

nicolo-ribaudo commented on packages/babel-plugin-transform-runtime/test/fixtures/absoluteRuntime/relative/subfolder/node_modules/@babel/runtime/helpers/classCallCheck.js:
```diff
@@ -0,0 +1 @@
+// empty
```
This empty file was needed to be able to `require.resolve` it.
